src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -269,10 +270,39 @@ const FabricInfo * RetrieveCurrentFabric(CommandHandler * aCommandHandler)
     return Server::GetInstance().GetFabricTable().FindFabricWithIndex(index);
 }
 
+void MatterRelatedDataCleanup()
+{
+    // Delete all user label data on the node which was added since it was commissioned.
+    DeviceLayer::DeviceInfoProvider * provider = DeviceLayer::GetDeviceInfoProvider();
+
+    if (provider)
+    {
+        for (auto endpoint : EnabledEndpointsWithServerCluster(OperationalCredentials::Id))
+        {
+            // If OperationalCredentials cluster is implemented on this endpoint
+            if (CHIP_NO_ERROR != provider->ClearUserLabelList(endpoint))
```

Comment:
Also, this does not make sense.  Why are we only clearing user labels on the endpoints that have Operational Credentials clusters on them (aka EP0 only)?

And why is this user label cleanup being done in the operational credentials server and not in the user label cluster impl?

Reply:
Yes, we should check against UserLabel cluster. The fabric is removed in  operational credentials server, how do we know the last fabric get deleted from user label server? Define a API and call from operational credentials server ? then we need to make sure is user label server is enabled from operational credentials server

Reply:
> The fabric is removed in operational credentials server, how do we know the last fabric get deleted from user label server?

The same way operational credentials does: register a delegate on the fabric table.  The operational credentials code being changed here is not the direct RemoveFabric code; it's just being notified by the fabric table....

Reply:
Acked